### PR TITLE
Kafka integration

### PR DESCRIPTION
**Pull request by Tomvbe**

### CODE REVIEW

File: ldi-core/kafka-common/pom.xml [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+        <version>0.0.1-SNAPSHOT</version>
+    </parent>
+
+    <artifactId>kafka-common</artifactId>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.kafka</groupId>
+            <artifactId>kafka-clients</artifactId>
+            <version>3.4.0</version>
```
Comment (WLefever-Cegeka) [inline ldi-core/kafka-common/pom.xml:18]: version

File: ldi-core/kafka-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/kafka/auth/SaslSslPlainConfigProvider.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+import java.util.Map;
+
+public class SaslSslPlainConfigProvider {
+
+	public Map<String, Object> createSaslSslPlainConfig(String user, String password) {
+		final Map<String, Object> properties = new HashMap<>();
+		properties.put(CommonClientConfigs.SECURITY_PROTOCOL_CONFIG, "SASL_SSL");
+		properties.put(SaslConfigs.SASL_MECHANISM, "PLAIN");
+		String plainLoginString = ("org.apache.kafka.common.security.plain.PlainLoginModule" +
+				" required username='%s' password='%s';").formatted(user, password);
```
Comment (WLefever-Cegeka) [inline ldi-core/kafka-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/kafka/auth/SaslSslPlainConfigProvider.java:16]: Suggestion: would put all strings here as constants.

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/pom.xml [hunk 3]
```diff
@@ -11,10 +11,25 @@
...
         </dependency>
+        <dependency>
+            <groupId>org.springframework.kafka</groupId>
+            <artifactId>spring-kafka-test</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>be.vlaanderen.informatievlaanderen.ldes.ldi</groupId>
+            <artifactId>kafka-common</artifactId>
+            <version>0.0.1-SNAPSHOT</version>
```
Comment (WLefever-Cegeka) [inline ldi-orchestrator/ldio-connectors/ldio-kafka-out/pom.xml:31]: project.version

File: ldi-core/kafka-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/kafka/auth/SaslSslPlainConfigProvider.java [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class SaslSslPlainConfigProvider {
+
+	public static final String PLAIN_LOGIN_BASE_STRING = "org.apache.kafka.common.security.plain.PlainLoginModule" +
+			" required username='%s' password='%s';";
+	public static final String SASL_SSL = "SASL_SSL";
+	public static final String PLAIN = "PLAIN";
```
Comment (WLefever-Cegeka) [inline ldi-core/kafka-common/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldi/kafka/auth/SaslSslPlainConfigProvider.java:14]: private if we don't use them elsewhere.

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutConfigurator.java
Comment (WLefever-Cegeka) [inline ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutConfigurator.java:1]: line 38 & 39 -> var

Comment (WLefever-Cegeka) [reply ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutConfigurator.java:1]: lline 65 as well

File: ldi-orchestrator/ldio-integration/src/test/java/be/vlaanderen/informatievlaanderen/ldes/KafkaIntegrationTest.java [hunk 5]
```diff
@@ -28,17 +30,34 @@
 import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertTrue;
 
-@EmbeddedKafka(brokerProperties = { "listeners=PLAINTEXT://localhost:9095", "port=9095" })
+//@EmbeddedKafka(brokerProperties = { "listeners=PLAINTEXT://localhost:9095", "port=9095" })
```
Comment (WLefever-Cegeka) [inline ldi-orchestrator/ldio-integration/src/test/java/be/vlaanderen/informatievlaanderen/ldes/KafkaIntegrationTest.java:33]: comment

File: ldi-orchestrator/ldio-integration/src/test/java/be/vlaanderen/informatievlaanderen/ldes/KafkaIntegrationTest.java [hunk 6]
```diff
@@ -28,17 +30,34 @@
...
+	@BeforeAll
+	static void foo() {
+		// embeddedKafkaBroker.kafkaPorts(9095);
+		embeddedKafkaBroker
+				.brokerProperty("listeners", "PLAINTEXT://%s".formatted("localhost:9095"))
+				.brokerProperty("port", 9095);
+		// .brokerProperty("security.inter.broker.protocol", "PLAINTEXT")
+		// .brokerProperty("sasl.enabled.mechanisms", "PLAIN")
+		// .brokerProperty("sasl.mechanism.inter.broker.protocol", "PLAIN");
+		// embeddedKafkaBroker.afterPropertiesSet();
```
Comment (WLefever-Cegeka) [inline ldi-orchestrator/ldio-integration/src/test/java/be/vlaanderen/informatievlaanderen/ldes/KafkaIntegrationTest.java:55]: comments

**Review by WLefever-Cegeka — APPROVED** [review]

**Comment by WLefever-Cegeka** [comment]
Noticed that there's some config duplication in the kafka in and out, maybe something to extract in the feature if possible.

File: ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutConfigurator.java
Comment (Tomvbe) [reply ldi-orchestrator/ldio-connectors/ldio-kafka-out/src/main/java/be/vlaanderen/informatievlaanderen/ldes/ldio/config/LdioKafkaOutConfigurator.java:1]: The vars is something we agreed on in a dev meeting, I believe they can improve readability in some cases

**Pull request merged**